src/include/migraphx/op/unary.hpp

```diff
@@ -84,10 +85,10 @@ struct unary : op_name<Derived>
         argument result{dyn_out.computed_shape};
         result.visit([&](auto output) {
             args[0].visit([&](auto input) {
-                std::transform(input.begin(),
-                               input.end(),
-                               output.begin(),
-                               static_cast<const Derived&>(*this).apply());
+                par_transform(input.begin(),
```

Comment:
How can any unary operation (i.e. calling std::transform here) be assumed to be parallelizable? Same question for binary operations using std::transform. Thanks.

Reply:
The `apply` cannot have side effects(unless you do bad things) because it only takes the implicit `this` as a parameter and that is already `const` since `compute` is a `const` method.

Reply:
OK. Could you please then add a comment here explaining why it is being used: that `apply` can't have side-effects, hence it is parallelizable. (And also to binary.hpp). Thanks.